### PR TITLE
ENT-2763 - Change packageOwnership type

### PR DESCRIPTION
**Pull request by tudor-malene**
Changes the type of the `packageOwnership` to avoid a bug in the old carpenter code

### CODE REVIEW

**Comment by shamsasari** [comment]
What's the benefit now of keeping `JavaPackageName`?

**Comment by tudor-malene** [comment]
> What's the benefit now of keeping `JavaPackageName`?

You're right. Not much benefit. I'll remove it.

File: core/src/main/kotlin/net/corda/core/node/NetworkParameters.kt [hunk 1]
```diff
@@ -199,34 +202,22 @@ data class NotaryInfo(val identity: Party, val validating: Boolean)
...
 
 // Check if a string is a legal Java package name.
 private fun isPackageValid(packageName: String): Boolean = packageName.isNotEmpty() && !packageName.endsWith(".") && packageName.split(".").all { token ->
     Character.isJavaIdentifierStart(token[0]) && token.toCharArray().drop(1).all { Character.isJavaIdentifierPart(it) }
 }
 
 // Make sure that packages don't overlap so that ownership is clear.
-private fun noOverlap(packages: Collection<JavaPackageName>) = packages.all { currentPackage ->
-    packages.none { otherPackage -> otherPackage != currentPackage && otherPackage.name.startsWith("${currentPackage.name}.") }
+private fun noOverlap(packages: Collection<String>) = packages.all { currentPackage ->
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/node/NetworkParameters.kt]: Can you move all these private methods and `owns` into `NetworkParameters` companion object

File: core/src/main/kotlin/net/corda/core/node/NetworkParameters.kt [hunk 2]
```diff
@@ -199,34 +202,22 @@ data class NotaryInfo(val identity: Party, val validating: Boolean)
...
-     * E.g.: "com.megacorp" owns "com.megacorp.tokens.MegaToken"
-     *
-     * Note: The ownership check is ignoring case to prevent people from just releasing a jar with: "com.megaCorp.megatoken" and pretend they are MegaCorp.
-     * By making the check case insensitive, the node will require that the jar is signed by MegaCorp, so the attack fails.
-     */
-    fun owns(fullClassName: String) = fullClassName.startsWith("$name.", ignoreCase = true)
-
-    override fun toString() = name
-}
+internal fun owns(packageName: String, fullClassName: String) = fullClassName.startsWith("$packageName.", ignoreCase = true)
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/node/NetworkParameters.kt]: `@CordaInternal`

File: core/src/main/kotlin/net/corda/core/node/NetworkParameters.kt [hunk 3]
```diff
@@ -111,6 +110,10 @@ data class NetworkParameters(
             Character.isJavaIdentifierStart(token[0]) && token.toCharArray().drop(1).all { Character.isJavaIdentifierPart(it) }
         }
 
+        fun requirePackageValid(name: String) {
```
Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/node/NetworkParameters.kt]: Can we move this to InternalUtils if other bits of the code need it.

**Pull request merged**